blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/RightclickManageServiceImpl.kt

```diff
@@ -54,19 +53,16 @@ class RightclickManageServiceImpl : RightclickManageService {
      * Executes the watcher for the given [player] if he has registered one.
      */
     override fun <P, L> executeWatchers(player: P, location: L): Boolean {
-        if (player !is Player) {
+        if (player !is Any) {
             throw IllegalArgumentException("Player has to be a BukkitPlayer!")
         }
 
         if (!rightClickListener.containsKey(player)) {
             return false
         }
 
-        if ((location as Location).block.state is Sign) {
-            val sign = location.block.state as Sign
-            sign.setLine(0, ChatColor.BOLD.toString() + "BlockBall")
-            sign.setLine(1, ChatColor.GREEN.toString() + "Loading...")
-            sign.update(true)
+        if (location is Any) {
```

Comment:
This if is pointless. 
Put proxyService.setSignLines outside of the if. Only if the result of proxyService.setSignLines is true, then call the rightClickListener statements.

Reply:
Check please!